src/endpoints/update.js

```diff
@@ -0,0 +1,114 @@
+const fs = require('fs');
+const Evilscan = require('evilscan');
+const promisify = require('util').promisify;
+const writeFile = promisify(fs.writeFile);
+const readFile = promisify(fs.readFile);
+const getThumbnails = require('../thumbnails').getThumbnails;
+
+/**
+ * Update hosts upon receiving get request for /update
+ */
+exports.get = async function (req, res) {
+  const config = JSON.parse(await readFile('configs/config.json'));
+  const portList = JSON.parse(await readFile('configs/ports.json'));
+  const options = {};
+  const ipRange = config.ipScanStart + '-' + config.ipScanEnd;
+  let ports = '';
+  const domain = config.domain;
+  const ipOmit = config.ipOmit;
+  const ipForce = config.ipForce;
+  const ignoreHost = config.ignoreHost;
+  let thumbnails = {};
+  const redirects = config.redirect;
+  const omitHosts = config.omitHosts;
+
+  global.allIps = {};
+
+  // scan for thumbnails
+  thumbnails = await getThumbnails();
+
+  // set the scanners options
+  options.target = ipRange;
+  for (const port in config.ports) {
+    if (port !== config.ports.length - 1) ports += config.ports[port] + ', ';
+    else ports += config.ports[port];
+  }
+
+  options.port = ports;
+  options.reverse = true;
+  options.json = true;
+
+  const results = [];
+  const scanner = new Evilscan(options);
+
+  // Run the scanner and parse results
+  scanner.on('result', function (data) {
+    global.allIps[data.ip] = data;
+    global.allIps[data.ip].omit = false;
+    global.allIps[data.ip].forced = false;
+    if (data.reverse) if (data.reverse.includes('.' + domain)) global.allIps[data.ip].reverse = global.allIps[data.ip].reverse.split('.')[0];
```

Comment:
Can you format the JS files? Aidan's code style is pretty trash and can be easily cleaned up lol

Reply:
rood

Reply:
i did format the js files, what do you mean?

Reply:
What formatter yields this as a single line?

`if (data.reverse) if (data.reverse.includes('.' + domain)) global.allIps[data.ip].reverse = global.allIps[data.ip].reverse.split('.')[0];`